sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java

```diff
@@ -338,10 +343,15 @@ public static BlobUrlParts parse(String url) {
     public static BlobUrlParts parse(URL url) {
         BlobUrlParts parts = new BlobUrlParts().setScheme(url.getProtocol());
 
-        if (ModelHelper.IP_V4_URL_PATTERN.matcher(url.getHost()).find()) {
-            parseIpUrl(url, parts);
-        } else {
-            parseNonIpUrl(url, parts);
+        try {
+            if (ModelHelper.determineAuthorityIsIpStyle(url.getAuthority())) {
+                parseIpUrl(url, parts);
+            } else {
+                parseNonIpUrl(url, parts);
+            }
+        } catch (MalformedURLException e) {
+            throw parts.logger.logExceptionAsError(new IllegalStateException("Authority is malformed. Host: "
```

Comment:
Should this be illegal state or illegal argument? isnt the url technically something the user provided to us?

Reply:
I think it should be IllegalState. The parameter type is URL, so that means we're starting with valid URL. The only way it's possibly malformed is if we constructed the dummy URL wrong, which is our fault. And given that we're using syntax that is known to be valid, if it somehow becomes malformed, I think it's fair to call the state illegal.